code/modules/mob/living/simple_animal/hostile/mimic.dm

```diff
@@ -266,3 +270,22 @@ GLOBAL_LIST_INIT(protected_objects, list(/obj/structure/table, /obj/structure/ca
 		return
 	icon_state = TrueGun.icon_state
 	icon_living = TrueGun.icon_state
+
+/datum/action/innate/mimic/root
+	name = "Root"
+	button_icon_state = "mimicA"
+	desc = "Roots yourself in place and disguises you as a normal crate. Anyone to try to open you up is given a nasty surprise!"
+
+/datum/action/innate/mimic/root/Activate()
+	var/mob/living/simple_animal/hostile/mimic/crate/S = owner
+
+	if(S.attempt_open == TRUE) //If we have tried to open it
+		S.visible_message("<span class='danger'><b>[S]</b> disguises itself as a crate</span>")
+
+		S.attempt_open = FALSE	//I WAS HIDING
+		S.icon_state = "crate"
+		S.icon_living = "crate"
+
+
+	//TBD, but see trellio
+	return
```

Comment:
a return at the end of a function that does nothing is a trailing return.
It's unneeded, and should be removed.